### DISCUSSION TITLE
Possible to set random parameter?

### DISCUSSION THREAD

**Discussion by ghost** [Vdauphin/HeartsAndMinds#1328] [Question and answer]
Hello!

Is it possible to create an array of factions and select one randomly on mission startup?

I was considering adding this to Params.hpp, so that we would have an automatic variety of factions on one mission file.

Also,

Are parameter settings saved to each mission database run? Or does it reset to default every time server restarts (but loads a save game).

Thanks!

**Comment by Vdauphin** [comment]
Hello,
It is possible just do something like selectRandom [your array]

> Or does it reset to default every time server restarts (but loads a save game)

Yes this. To be accurate it depends on setting. The enemy density parameter is used when you enter the first time in a city. So if you change this setting between two save, cities activated Will not be affected but city newly activate will check this setting

**Reply by ghost** [reply]
So if I were to use;

_faction = selectRandom [x, y, z];

In Params.hpp, then it would potentially change faction each time I restart? Or would it use whatever was saved in the database? 

Thanks

**Reply by Vdauphin** [reply]
You can't do it in param.hpp
You must use mission.sqf 

>  then it would potentially change faction each time I restart?

Yes exactly. For city previously activated, soldier will be thé same as before the param change but city not activated the New param will be used

You need to save thé faction name and load it at each server start

**Reply by ghost** [reply]
Got it, thanks for the info!
